jbpm-human-task/jbpm-human-task-workitems/src/test/java/org/jbpm/services/task/wih/util/HumanTaskHandlerHelperTest.java

```diff
@@ -137,130 +304,986 @@ public void testSetDeadlinesNotStartedReassignTwoTimes() {
-		assertEquals(1, deadlines.getEndDeadlines().get(0).getEscalations().size());
-		assertEquals(1, deadlines.getEndDeadlines().get(0).getEscalations().get(0).getReassignments().size());
-		assertEquals(0, deadlines.getEndDeadlines().get(0).getEscalations().get(0).getNotifications().size());
-		
+		assertEquals(2, deadlines.getStartDeadlines().size());
+		assertEquals(0, deadlines.getEndDeadlines().size());
+		assertEquals(1, deadlines.getStartDeadlines().get(0).getEscalations().size());
+		assertEquals(1, deadlines.getStartDeadlines().get(0).getEscalations().get(0).getReassignments().size());
+		assertEquals(0, deadlines.getStartDeadlines().get(0).getEscalations().get(0).getNotifications().size());
+
+		assertEquals(1, deadlines.getStartDeadlines().get(1).getEscalations().size());
+		assertEquals(1, deadlines.getStartDeadlines().get(1).getEscalations().get(0).getReassignments().size());
+		assertEquals(0, deadlines.getStartDeadlines().get(1).getEscalations().get(0).getNotifications().size());
+
 		// verify reassignment
-		Reassignment reassignment = deadlines.getEndDeadlines().get(0).getEscalations().get(0).getReassignments().get(0);
-		assertEquals(2, reassignment.getPotentialOwners().size());
+		Reassignment reassignment = deadlines.getStartDeadlines().get(0).getEscalations().get(0).getReassignments().get(0);
+		assertEquals(1, reassignment.getPotentialOwners().size());
 		assertEquals("john", reassignment.getPotentialOwners().get(0).getId());
-		assertEquals("sales", reassignment.getPotentialOwners().get(1).getId());
-		
+
 		// check deadline expiration time
-		assertNotNull(deadlines.getEndDeadlines().get(0).getDate());
-		long expirationTime = deadlines.getEndDeadlines().get(0).getDate().getTime() - System.currentTimeMillis();
-		
+		assertNotNull(deadlines.getStartDeadlines().get(0).getDate());
+		long expirationTime = deadlines.getStartDeadlines().get(0).getDate().getTime() - System.currentTimeMillis();
+
 		assertEquals(4, roundExpirationTime(expirationTime));
-		
+
+		// verify reassignment
+		reassignment = deadlines.getStartDeadlines().get(1).getEscalations().get(0).getReassignments().get(0);
+		assertEquals(1, reassignment.getPotentialOwners().size());
+		assertEquals("john", reassignment.getPotentialOwners().get(0).getId());
+
+		// check deadline expiration time
+		assertNotNull(deadlines.getStartDeadlines().get(1).getDate());
+		expirationTime = deadlines.getStartDeadlines().get(1).getDate().getTime() - System.currentTimeMillis();
+
+		assertEquals(6, roundExpirationTime(expirationTime));
+
 	}
-	
+
 	@Test
-	public void testSetDeadlinesNotCompletedReassignTwoTimes() {
-		
+	public void testSetDeadlinesNotStartedReassignTwoTimesWithISOExpirationTimePeriodFormatWithRepeatCount() {
 		WorkItem workItem = new WorkItemImpl();
-		workItem.setParameter("NotCompletedReassign", "[users:john]@[4h,6h]");
-		
+		workItem.setParameter("NotStartedReassign", "[users:john]@[R3/PT4H,R/PT6H]");
```

Comment:
This is a strange one... you are setting one limited repetition and the other one unbounded repetition... does not look strange to you ?